### DISCUSSION TITLE
Help merging two images

### DISCUSSION THREAD

**Discussion by mlavespere** [ImageMagick/ImageMagick#2211] [Help]
I using v6.9 and need some help merging two images. The images are the same size and resolution. Image1 has a large black area which I would like to fill with the corresponding area of Image2. I'm using "composite -dissolve 75 image1.jpg image2.jpg merged.jpg" to get close but I really need to mask out the area of Image1 that isn't black.

Thanks.

**Comment by fmw42** [comment]
Please post you input images and your merged image so we can see what you are trying to do.

**Reply by mlavespere** [reply]
Done. I'd like image2 to appear only where image1 is black. Thanks.

![merge](https://user-images.githubusercontent.com/16197311/86052156-84556780-ba24-11ea-9ee1-07cd5572778c.jpg)
![image1](https://user-images.githubusercontent.com/16197311/86052169-88818500-ba24-11ea-94aa-8e86d51e3072.jpg)
![image2](https://user-images.githubusercontent.com/16197311/86052172-8b7c7580-ba24-11ea-8e99-d570f92e081d.jpg)

**Reply by fmw42** [reply]
You show 3 images. Which is image1 and which is image2? Is the third one the result?

Try (unix syntax)

```
convert image1.jpg image2.jpg \
\( -clone 0 -threshold 0 -negate \) \
-compose over -composite result.jpg
```


![result](https://user-images.githubusercontent.com/13458218/86054042-48170c80-ba0e-11ea-8921-3f2772e26970.jpg)



Or

```
convert image1.jpg image2.jpg \
\( -clone 0 -threshold 0 -negate \) \
-compose over -composite -gamma 2 result2.jpg
```


![result2](https://user-images.githubusercontent.com/13458218/86054330-cd9abc80-ba0e-11ea-99b0-66bd00616702.jpg)

**Reply by mlavespere** [reply]
Sorry. I thought it would show the file names. The first is the merged image, the 2nd has big black area, and the third is what I'd like to fill in the black area with. I'll give your commands a try. Thanks.

**Reply by fmw42** [reply]
Yes. I used the wrong images.  I do not know what filenames you used. The images I downloaded had very long rather hideous names.

**Reply by mlavespere** [reply]
Thanks for the help. The -gamma option helped a lot.
